flutter_todo/lib/screens/log_in.dart

```diff
@@ -30,133 +30,81 @@ class _LogInState extends State<LogIn> {
-                            _errorMessage = null;
-                          }
-                        });
-                      },
-                      decoration: const InputDecoration(
-                          border: OutlineInputBorder(),
-                          labelText: 'Password',
-                          hintText: 'Enter secure password'),
-                    ),
-                  ),
-                  Container(
-                    height: 50,
-                    width: 250,
-                    margin: const EdgeInsets.symmetric(vertical: 25),
-                    decoration: BoxDecoration(
-                        color: Colors.blue,
-                        borderRadius: BorderRadius.circular(20)),
-                    child: TextButton(
-                      onPressed: () {
-                        if (_logIn) {
-                          _logInUser();
-                        } else {
-                          _signUpUser();
-                        }
-                      },
-                      child: Text(
-                        _logIn ? 'Log in' : 'Sign up',
-                        style:
-                            const TextStyle(color: Colors.white, fontSize: 25),
-                      ),
-                    ),
+                    _isLogin ? "Log in" : "Sign up",
+                    style: const TextStyle(color: Colors.white, fontSize: 20),
                   ),
-                ],
-              )),
-              Container(
-                  padding: const EdgeInsets.symmetric(vertical: 25),
-                  child: _errorMessage != null
-                      ? Text(_errorMessage!,
-                          style: const TextStyle(color: Colors.red))
-                      : null),
+                ),
+              ),
               TextButton(
-                  onPressed: _setLogInSignUpState,
+                  onPressed: () => setState(() => _isLogin = !_isLogin),
                   child: Text(
-                    _logIn
-                        ? "New to Flutter Realm Todo? Sign up"
-                        : 'Already have an account? Log in.',
-                  ))
+                    _isLogin ? "New to Flutter Realm Todo? Sign up" : 'Already have an account? Log in.',
+                  )),
+              Padding(
+                padding: const EdgeInsets.all(25),
+                child: Text(_errorMessage ?? "", style: const TextStyle(color: Colors.red)),
+              ),
             ],
           ),
+          ),
         ),
       ),
     );
   }
+
+  Widget loginField(TextEditingController controller, {String? labelText, String? hintText, bool? obscure}) {
+    return Padding(
+      padding: const EdgeInsets.all(15),
+      child: TextField(
+          obscureText: obscure ?? false,
+          controller: controller,
+          decoration: InputDecoration(border: const OutlineInputBorder(), labelText: labelText, hintText: hintText)),
+    );
+  }
```

Comment:
should this be in the widgets.dart file?

Reply:
Yes it could be. I left it here, because it is about the login. But ok I will move this. Fixed here https://github.com/mongodb-university/realm-template-apps/pull/110/commits/b7e0b60ddfec5e3bd775f11b6002c01d3d714c55